src/modules/commandrunner/src/lib/Command.h

```diff
@@ -0,0 +1,143 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#ifndef COMMAND_H
+#define COMMAND_H
+
+#include <cstring>
+#include <mutex>
+#include <rapidjson/document.h>
+#include <rapidjson/stringbuffer.h>
+#include <rapidjson/writer.h>
+#include <string>
+
+#include <CommonUtils.h>
+#include <Logging.h>
+
+const std::string g_commandArguments = "CommandArguments";
+const std::string g_commandId = "CommandId";
+const std::string g_arguments = "Arguments";
+const std::string g_action = "Action";
+const std::string g_timeout = "Timeout";
+const std::string g_singleLineTextResult = "SingleLineTextResult";
+
+const std::string g_commandStatus = "CommandStatus";
+const std::string g_resultCode = "ResultCode";
+const std::string g_textResult = "TextResult";
+const std::string g_currentState = "CurrentState";
+
+#define COMMANDRUNNER_LOGFILE "/var/log/osconfig_commandrunner.log"
+#define COMMADRUNNER_ROLLEDLOGFILE "/var/log/osconfig_commandrunner.bak"
+
+class CommandRunnerLog
+{
+public:
+    static OSCONFIG_LOG_HANDLE Get()
+    {
+        return m_log;
+    }
+
+    static void OpenLog()
+    {
+        m_log = ::OpenLog(COMMANDRUNNER_LOGFILE, COMMADRUNNER_ROLLEDLOGFILE);
+    }
+
+    static void CloseLog()
+    {
+        ::CloseLog(&m_log);
+    }
+
+    static OSCONFIG_LOG_HANDLE m_log;
+};
+
+class Command
+{
+public:
+    enum Action
+    {
+        None = 0,
+        Reboot,
+        Shutdown,
+        RunCommand,
+        RefreshCommandStatus,
+        CancelCommand
+    };
+
+    enum State
+    {
+        Unknown = 0,
+        Running,
+        Succeeded,
+        Failed,
+        TimedOut,
+        Canceled
+    };
+
+    class Arguments
+    {
+    public:
+        const std::string id;
+        const std::string command;
```

Comment:
Using consistent name matching with the MIM could be useful. "commandId",  "arguments" (arguments string is implied on both Windows and Linux to contain the command line)